### DISCUSSION TITLE
TimeInput support for 24 hours?

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#5498] [Q&A] [answered]
Currently `TimeInput` can accept hour 01-12 with AM/PM.
Is there anyway we can input 01-24?

**Comment by rtivital** [comment, marked as answer]
TimeInput is wrapper around `input[type=time]`, the format is based on user language.

**Reply by Toshinaki** [reply]
So I can control the behavior by setting `<html lang="xx"`?
Thanks.

**Reply by butterfly-valley** [reply]
Hello, setting html lang does NOT do the trick, it still is 12h, how can we manipulate it? Thanks.

**Reply by kolmssi** [reply]
Any possibilities to add possibility to force the format for all users?

**Reply by rtivital** [reply]
This feature will be available in 8.0 release that is scheduled to be released next year.

**Reply by ronangaillard** [reply]
What is a hot fix we can use right now ? We have lot of issues with users having a 24h locale set on their computer/browser but seeing 12h formatted time.

**Reply by ronangaillard** [reply]
For example on Firefox 134.0.2

<img width="683" alt="image" src="https://github.com/user-attachments/assets/ef03f342-6db8-4e7f-8cef-5afc34dbf559" />
<img width="531" alt="image" src="https://github.com/user-attachments/assets/b9b8d896-7e83-45e0-a7f4-049a48947038" />

**Reply by srtp** [reply]
> TimeInput is wrapper around `input[type=time]`, the format is based on user language.

Pls add in mantine v6
